### PR TITLE
Refactor release issue template and release wiki based on new changes

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Refactor release issue template and release wiki based on new changes

### Issues Resolved
#4908 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — COMMENTED** [review]
There are more under the folded `<details><summary>Release Checklist</summary>`.

**Comment by gaiksaya** [comment]
> There are more under the folded `<details><summary>Release Checklist</summary>`.

Yeah wondering if they are needed or not.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 1]
```diff
@@ -145,14 +141,6 @@ Release blog is ready | :red_circle: |   |
 - [ ] Advertise on Social Media.
 - [ ]  Post on public slack and Github Release issue.
 
-### Post-Release activities
```
Comment (prudhvigodithi) [inline .github/ISSUE_TEMPLATE/release_template.md:148]: We still need these right @gaiksaya ?
For Release Tags and Input Manifest Update we have an automated workflow which we can add here. Also for Helm and Ansible Playbook release. today we do them after the release.

Comment (gaiksaya) [reply .github/ISSUE_TEMPLATE/release_template.md:148]: It was a duplicate places in checklist. See https://github.com/opensearch-project/opensearch-build/pull/5031/files#diff-38e15f3a510b3e0a8b537c78468a4c2db29c2f00c55abbd0a310fbcfe9c395c2R148

**Comment by gaiksaya** [comment]
Final template can be viewed here: https://github.com/opensearch-project/opensearch-build/blob/e250a2cab018ebc3162eaf693d25539213560e32/.github/ISSUE_TEMPLATE/release_template.md

**Review by peterzhuamazon — APPROVED** [review]

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**